osv/models.py

```diff
@@ -364,7 +364,18 @@ def _tokenize(self, value):
       return []
 
     value_lower = value.lower()
-    return re.split(r'\W+', value_lower) + [value_lower]
+    tokens = {token for token in re.split(r'\W+', value_lower) if token}
+    tokens.add(value_lower)
+
+    parts = value_lower.split('-')
```

Comment:
Can you add some code comments here to explain what this is doing?

Reply:
An example might also be helpful
```
e.g. `a-b-c-d' becomes ['a-b', 'b-c', 'c-d', 'a-b-c', 'b-c-d', 'a-b-c-d']
```

Reply:
I think there's an existing functools function to do this.